datacapturing/src/main/java/de/cyface/datacapturing/DataCapturingService.java

```diff
@@ -791,28 +790,72 @@ private void stopServiceSync(final long timeout, final @NonNull TimeUnit unit)
-        context.registerReceiver(finishedHandler, new IntentFilter(MessageCodes.BROADCAST_SERVICE_STOPPED));
+        Log.v(TAG, "Registering finishedHandler for service stop synchronization broadcast.");
+        context.registerReceiver(finishedHandler,
+                new IntentFilter(MessageCodes.FINISHED_HANDLER_BROADCAST_SERVICE_STOPPED));
 
-        Log.v(TAG, String.format("Stopping using Intent with context %s", context));
-        final Intent stopIntent = new Intent(context, DataCapturingBackgroundService.class);
-        final boolean serviceWasActive = context.stopService(stopIntent);
+        boolean serviceWasActive;
+        try {
+            // For some reasons we have to call the unbind here.
+            // We tried to send the stopIntent to the BackgroundService first which calls it's onDestroy
+            // method which sends an SERVICE_STOPPED message back to this service where we executed
+            // the unbind method. This should have worked for both - stopping the BackgroundService from
+            // this service and by itself (via eventHandlerStrategyImpl.handleSpaceWarning())
+            unbind();
+        } catch (IllegalArgumentException e) {
+            throw new DataCapturingException(e);
+        } catch (DataCapturingException e) {
+            Log.w(TAG, "Service was either paused or already stopped, so I was unable to unbind from it.");
+        } finally {
+            Log.v(TAG, String.format("Stopping using Intent with context %s", context));
+            Intent stopIntent = new Intent(context, DataCapturingBackgroundService.class);
+            serviceWasActive = context.stopService(stopIntent);
+        }
 
         if (!serviceWasActive) {
-            // The background service was not running so we need to inform the caller of this method ourselves.
-            Intent stoppedBroadcastIntent = new Intent(MessageCodes.BROADCAST_SERVICE_STOPPED);
-            stoppedBroadcastIntent.putExtra(STOPPED_SUCCESSFULLY, false);
-            context.sendBroadcast(stoppedBroadcastIntent);
+            // The background service was *not* running so we need to inform the caller of this method ourselves.
+            // This can happen when we stop a "paused" DataCapturingBackgroundService (which is already stopped).
+            sendServiceStoppedUnsuccessfullyCallback(context);
         }
     }
 
+    /**
+     * This message is sent to the {@link ShutDownFinishedHandler} to inform callers that the async stop
+     * command was executed and that the {@link DataCapturingBackgroundService} was not active when the
+     * stop command was received which can happen when the pause command was executed before.
+     *
+     * @param context The {@link Context} used to send the broadcast from
+     */
+    private void sendServiceStoppedUnsuccessfullyCallback(final Context context) {
+        sendServiceStoppedBroadcast(context, -1, false);
```

Comment:
If we stop a paused service, there should still have been an open measurement (which should be closed by the current call to stop) and thus we should no have to use -1 here.